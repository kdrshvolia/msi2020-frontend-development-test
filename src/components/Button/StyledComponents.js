import styled, { css } from 'styled-components';

const SearchButton = styled.button`
  cursor: pointer;
  width: 152px;
  height: 42px;
  background: linear-gradient(92.01deg, #8ea7ff 0%, #7291ff 100%);
  border-radius: 10px;
  font-family: Roboto;
  font-style: normal;
  font-weight: bold;
  font-size: 16px;
  line-height: 22px;
  color: #ffffff;
  border: none;
`;

export { SearchButton };
