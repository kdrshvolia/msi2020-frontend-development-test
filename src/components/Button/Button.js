import React from 'react';import PropTypes from 'prop-types';import { SearchButton } from './StyledComponents';const Button = ({ text, onClick }) => {  const handleClick = () => {    onClick();  };  return (    <div>      <SearchButton type="button" onClick={handleClick}>        {text}      </SearchButton>    </div>  );};Button.propTypes = {  text: PropTypes.string.isRequired,  onClick: PropTypes.func.isRequired,};export default Button;