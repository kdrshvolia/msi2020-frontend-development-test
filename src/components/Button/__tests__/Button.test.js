import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Button from '../Button';

describe('Button', () => {
  it('Button is clicked', () => {
    const text = 'Button';
    const isClicked = jest.fn();
    const wrapper = mount(<Button onClick={isClicked} text={text} />);
    const button = wrapper.find('button');
    button.simulate('click');
    expect(isClicked).toBeCalledWith();
  });
  it('Reders correctly', () => {
    const text = 'Button';
    const rendered = renderer.create(<Button text={text} />);
    expect(rendered).toMatchSnapshot();
  });
});
