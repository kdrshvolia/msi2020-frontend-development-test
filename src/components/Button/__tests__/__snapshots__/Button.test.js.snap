// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Reders correctly 1`] = `
<div>
  <button
    className="sc-AxjAm gljxek"
    onClick={[Function]}
    type="button"
  >
    Button
  </button>
</div>
`;
